### ISSUE TITLE
StyleSpace Analysis: Disentangled Controls for StyleGAN Image Generation

### ISSUE THREAD

**Issue by reyllama** [reyllama/paper-reviews#121]
*CVPR 2021*

# TL;DR

- Style parameters space is more disentangled than W or W+ space.
- We can gain local and attribute specific control on generated images through style space exploration.

# Motivation

- Disentanglement is the key to semantic image editing.
- StyleGAN has been shown to have good latent disentanglement quality thanks to augmented latent space W.
- Style parameters are key driving factor of styleGAN's success, but analysis on style space has not been made enough.

# Previous Work

- GANspace
- InterFaceGAN

# Method

## Style Parameters
![image](https://user-images.githubusercontent.com/46308407/131446143-55815f1c-6db5-44c7-a8e9-26b90f6d4444.png)

- Stylespace is quantitatively shown to be more disentangled.

## Local Control

- Compute channel-wise image gradient and compare with pretrained segmentation network output (=semantic map)
- The channels that overlap with specific category (ex: beard) consistently is believed to ***control the semantic***.

![image](https://user-images.githubusercontent.com/46308407/131446855-f5306f93-aaa7-44f9-924f-50c62735cc1c.png)
![image](https://user-images.githubusercontent.com/46308407/131446861-4d1c0f2c-2214-48cf-8158-11997e97b37c.png)

![image](https://user-images.githubusercontent.com/46308407/131446967-fffcb50f-9c5a-4422-b436-4f0b349b3fbb.png)

## Attribute Control
- Use 10-30 exemplar images with the attribute of interest.
- Compute mean difference with the population for each channel, and find the one that deviates most from the population mean.

![image](https://user-images.githubusercontent.com/46308407/131446987-224a5cb9-9d6b-49af-bcc8-9fe115bfb652.png)

## Real Image Editing
- Use an encoder to find the latent corresponding to an image.
![image](https://user-images.githubusercontent.com/46308407/131447067-7bbf8f0b-0c5f-4554-94d0-3f4e6e32b52f.png)

## Disentanglement

![image](https://user-images.githubusercontent.com/46308407/131447197-ef3da108-a2b3-462c-86d1-6771ebb7face.png)

## Attribute Informativeness and Comparison

![image](https://user-images.githubusercontent.com/46308407/131447104-58f1f9ff-4375-40a8-8105-b61087880187.png)
![image](https://user-images.githubusercontent.com/46308407/131447119-a8c4446c-54f7-451b-a642-c6273f9b12b9.png)

## toRGB layers parameters
![image](https://user-images.githubusercontent.com/46308407/131447253-08c030df-e20f-466a-907f-5eaee0c506e5.png)

- Parameters in toRGB layers control the overall tone of the image, hence not directly manipulatable for editing purpose.

# Comments

- Many papers on styleGAN latent analysis and editing.
- StyleGAN possess many beneficial properties such as disentanglement and style control.
- On top of that, various normalizations (ISLA-Norm, Region-Aware Instance Normalization, SPADE, etc) are proven to be an effective means of injecting style or controlling the image.
- Fine grained control shown in the paper is impressive, but considering that the dataset is highly structured and constrained, I think the research interest has to move to composing more complex scenes and images in the wild.
